cpp/src/arrow/compute/exec/aggregate.cc

```diff
@@ -61,7 +61,11 @@ Result<std::vector<std::unique_ptr<KernelState>>> InitKernels(
       // use known default options for the named function if possible
       auto maybe_function = ctx->func_registry()->GetFunction(aggregates[i].function);
       if (maybe_function.ok()) {
-        options = maybe_function.ValueOrDie()->default_options();
+        const FunctionOptions* default_opts =
```

Comment:
I think if you change line 58 to...

```
const FunctionOptions* options = aggregates[i].options.get();
```

...then this change isn't needed.  We are just grabbing a read-only view of the options which we can safely discard after the call to `init`

Reply:
Good catch. Thank you.